src/main/java/lz/modelTesting/controllers/MockupsController.java

```diff
@@ -0,0 +1,52 @@
+package lz.modelTesting.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lz.modelTesting.documents.Mockup;
+import lz.modelTesting.repositories.MockupsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.util.List;
+
+@RestController
+@RequestMapping("/mockups")
+public class MockupsController {
+
+    @Autowired
+    private MockupsRepository mockupsRepository;
+
+    @GetMapping(value = "/all")
+    public List<Mockup> getAll() {
+        return mockupsRepository.findAll();
+    }
+
+    @GetMapping(value = "/name/{name}")
+    public ResponseEntity<String> getByName(@PathVariable String name) throws JsonProcessingException {
+        Mockup res = mockupsRepository.findByName(name).get(0);
```

Comment:
Changed method to return 404 in case of NoSuchElementException

Reply:
don't use `Optional.get()` without `isPresent()`
it may produce exception you actually handled but it would be better to avoid it.
Maybe something like that? Feel free to do it another way
```
    public ResponseEntity<String> getById(@PathVariable String id) {
        try {
            Optional<Mockup> mockup = mockupsRepository.findById(id);
            
            return mockup.map(this::wrapMockupToString)
                    .orElseThrow()
                    .map(mockupString -> ResponseEntity.ok().body(mockupString))
                    .orElse(ResponseEntity.notFound().build());
            
        } catch (Exception ex) {
            return ResponseEntity.badRequest().build();
        }
    }

    private Optional<String> wrapMockupToString(Mockup mockup) {
        try {
            return Optional.of(objectMapper.writeValueAsString(mockup));
        } catch (JsonProcessingException e) {
            return Optional.empty();
        }
    }
```

Reply:
Changed as above